scripts/pkglib.sh

```diff
@@ -164,6 +164,33 @@ function checkout_repo()
     cp ../scripts/apport_hook.py $OBS_WORKDIR/$APPORT_HOOK
 }
 
+function generate_kernel_config() {
+    local kernel_version=$1
+    local kernel_filename=$2
+    local kernel_url=https://www.kernel.org/pub/linux/kernel/v4.x/linux-"${kernel_version}".tar.xz
+
+    local tmpdir=/var/tmp
+    local kernel_tree_dir="${tmpdir}/linux-${kernel_version}"
+
+    # Download the kernel source tree tarball and decompress it in ${tmpdir}
+    [ ! -e "${kernel_tree_dir}.tar.xz" ] &&
+        curl -o "${kernel_tree_dir}.tar.xz" -L $kernel_url || true
```

Comment:
Could you add a variable to avoid hard-coding the `.tar.xz` in various places? That will make it easier and safer to update this script when someone invents a better compression algorithm ;)